examples/toolbox-weather-event.php

```diff
@@ -23,12 +26,22 @@
 $platform = PlatformFactory::create($_ENV['OPENAI_API_KEY']);
 $llm = new GPT(GPT::GPT_4O_MINI);
 
-$wikipedia = new OpenMeteo(HttpClient::create());
-$toolBox = new ToolBox(new ToolAnalyzer(), [$wikipedia]);
-$processor = new ChainProcessor($toolBox);
+$openMeteo = new OpenMeteo(HttpClient::create());
+$toolBox = new ToolBox(new ToolAnalyzer(), [$openMeteo]);
+$eventDispatcher = new EventDispatcher();
+$processor = new ChainProcessor($toolBox, eventDispatcher: $eventDispatcher);
 $chain = new Chain($platform, $llm, [$processor], [$processor]);
 
-$messages = new MessageBag(Message::ofUser('How is the weather currently in Berlin? And how about tomorrow?'));
+// Add tool call result listener to enforce chain exits direct with structured response for weather tools
+$eventDispatcher->addListener(ToolCallExecuted::class, function (ToolCallExecuted $event): void {
+    foreach ($event->toolCallResults as $toolCallResult) {
```

Comment:
The event name is singular and then you iterate over all toolCallResults (plural), that feels weird. Shouldn't we throw an event for every ToolCall which was executed?

Otherwise rename it to `ToolCallsExecuted`?

Or am I missing sth.?

Reply:
you're right. should be plural